src/components/interactive/dragBoxLayer.ts

```diff
@@ -0,0 +1,322 @@
+///<reference path="../../reference.ts" />
+
+module Plottable {
+export module Component {
+  export module Interactive {
+    type _EdgeIndicator = {
+      top: boolean;
+      bottom: boolean;
+      left: boolean;
+      right: boolean;
+    }
+
+    export class DragBoxLayer extends Component.SelectionBoxLayer {
+      private _dragInteraction: Interaction.Drag;
+      private _detectionEdgeT: D3.Selection;
+      private _detectionEdgeB: D3.Selection;
+      private _detectionEdgeL: D3.Selection;
+      private _detectionEdgeR: D3.Selection;
+      private _detectionCornerTL: D3.Selection;
+      private _detectionCornerTR: D3.Selection;
+      private _detectionCornerBL: D3.Selection;
+      private _detectionCornerBR: D3.Selection;
+
+      private _detectionRadius = 3;
+      private _xResizable = false;
+      private _yResizable = false;
+
+      private _dragStartCallback: (b: Bounds) => any;
+      private _dragCallback: (b: Bounds) => any;
+      private _dragEndCallback: (b: Bounds) => any;
+
+      constructor() {
+        super();
+        this.classed("drag-box-layer", true);
+
+        this._dragInteraction = new Interaction.Drag();
+        this._setUpCallbacks();
+        this.registerInteraction(this._dragInteraction);
+      }
+
+      private _setUpCallbacks() {
+        var grabbedEdges: _EdgeIndicator;
+        var topLeft: Point;
+        var bottomRight: Point;
+
+        this._dragInteraction.onDragStart((s: Point) => {
+          grabbedEdges = this._getEdges(s);
+          if (!this._yResizable) {
+            grabbedEdges.top = false;
+            grabbedEdges.bottom = false;
+          }
+          if (!this._xResizable) {
+            grabbedEdges.left = false;
+            grabbedEdges.right = false;
+          }
+
+          if (!this.boxVisible() ||
+              (!grabbedEdges.top && !grabbedEdges.bottom &&
+               !grabbedEdges.left && !grabbedEdges.right)
+             ) {
+            this.bounds({
+              topLeft: s,
+              bottomRight: s
+            });
+            // a new box is effectively "grabbed" at the bottom-right corner
+            grabbedEdges.bottom = true;
+            grabbedEdges.right = true;
+          }
+
+          this.boxVisible(true);
+          var bounds = this.bounds();
+          // copy points so changes to topLeft and bottomRight don't mutate bounds
+          topLeft = { x: bounds.topLeft.x, y: bounds.topLeft.y };
+          bottomRight = { x: bounds.bottomRight.x, y: bounds.bottomRight.y };
+          if (this._dragStartCallback) {
+            this._dragStartCallback(bounds);
+          }
+        });
+
+        this._dragInteraction.onDrag((s: Point, e: Point) => {
+          if (grabbedEdges.bottom) {
+            bottomRight.y = e.y;
+          } else if (grabbedEdges.top) {
+            topLeft.y = e.y;
+          }
```

Comment:
Yes, but it's only the point that was initially in the bottom-right (hence the original name `_initBR`).